string in_msg
---
string out_msg
